### PR TITLE
Add OpenStack Designate DNS API support

### PR DESCRIPTION
**Pull request by andybotting**
This provider relies on the the python-openstackclient and
python-designateclient packages be installed and available
in your path, with either password or application credentials
loaded in your environment.

Any issues should be reported at:
https://github.com/acmesh-official/acme.sh/issues/3054

### CODE REVIEW

File: dnsapi/openstack.sh [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+########  Public functions #####################
+
+# Usage: dns_openstack_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_openstack_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+
+  _dns_openstack_credentials || return $?
```
Comment (Neilpang) [inline dnsapi/openstack.sh:22]: did you forget to check "_exists openstack"  ?

Comment (andybotting) [reply dnsapi/openstack.sh:22]: It's called in the `_dns_openstack_check_setup` function. I separated it into a function as I may want to add more setup checks in the future

**Comment by Neilpang** [comment]
please fix the CI errors: 
https://travis-ci.org/github/acmesh-official/acme.sh/jobs/708539530

**Comment by andybotting** [comment]
Looks like a CI check issue got merged from my earlier deploy PR. Do you want that fixed as a separate commit?

**Comment by Neilpang** [comment]
Yes,  go ahead

**Comment by andybotting** [comment]
Fix added in https://github.com/acmesh-official/acme.sh/pull/3059

**Pull request merged**